Gems/AudioSystem/Code/Source/Engine/AudioSystem.h

```diff
@@ -111,53 +91,36 @@ namespace Audio
         bool ReleaseAudioListenerID(const TAudioObjectID nAudioObjectID) override;
         bool SetAudioListenerOverrideID(const TAudioObjectID nAudioObjectID) override;
 
-        void GetInfo(SAudioSystemInfo& rAudioSystemInfo) override;
         const char* GetControlsPath() const override;
         void UpdateControlsPath() override;
         void RefreshAudioSystem(const char* const levelName) override;
 
-        IAudioProxy* GetFreeAudioProxy() override;
-        void FreeAudioProxy(IAudioProxy* const pIAudioProxy) override;
+        IAudioProxy* GetAudioProxy() override;
+        void RecycleAudioProxy(IAudioProxy* const pIAudioProxy) override;
 
         TAudioSourceId CreateAudioSource(const SAudioInputConfig& sourceConfig) override;
         void DestroyAudioSource(TAudioSourceId sourceId) override;
 
-        // When AUDIO_RELEASE is defined, these two functions always return nullptr
-        const char* GetAudioControlName(const EAudioControlType controlType, const TATLIDType atlID) const override;
-        const char* GetAudioSwitchStateName(const TAudioControlID switchID, const TAudioSwitchStateID stateID) const override;
-
-    protected:
-        void ProcessRequestThreadSafe(CAudioRequestInternal audioRequestInternalData);
-
     private:
-        using TAudioRequests = AZStd::deque<CAudioRequestInternal, Audio::AudioSystemStdAllocator>;
         using TAudioProxies = AZStd::vector<CAudioProxy*, Audio::AudioSystemStdAllocator>;
 
         void InternalUpdate();
-        bool ProcessRequests(TAudioRequests& rRequestQueue);
-        void ProcessRequestBlocking(CAudioRequestInternal& audioRequestInternalData);
-
-        void ExecuteRequestCompletionCallbacks(TAudioRequests& requestQueue, AZStd::mutex& requestQueueMutex, bool bTryLock = false);
-        void ExtractCompletedRequests(TAudioRequests& rRequestQueue, TAudioRequests& rSyncCallbacksQueue);
 
         bool m_bSystemInitialized;
 
         using duration_ms = AZStd::chrono::duration<float, AZStd::milli>;
-        const duration_ms m_targetUpdatePeriod = AZStd::chrono::milliseconds(8);
+        const duration_ms m_targetUpdatePeriod = AZStd::chrono::milliseconds(4);
 
         CAudioTranslationLayer m_oATL;
         CAudioThread m_audioSystemThread;
 
-
-        TAudioRequests m_blockingRequestsQueue;     // blocking requests go here, main thread will wait for audio thread to process
-        TAudioRequests m_threadSafeCallbacksQueue;  // requests coming from any thread go here.
-        TAudioRequests m_pendingCallbacksQueue;     // this queue holds pending callbacks, agreggated from processed requests
-
-        AZStd::mutex m_blockingRequestsMutex;
-        AZStd::mutex m_threadSafeCallbacksMutex;
+        AudioRequestsQueue m_blockingRequestsQueue;
+        AudioRequestsQueue m_pendingRequestsQueue;
+        AudioRequestsQueue m_pendingCallbacksQueue;
+        AZStd::recursive_mutex m_blockingRequestsMutex;
+        AZStd::mutex m_pendingRequestsMutex;
```

Comment:
Any reason why the blocking request mutex is a recursive mutex, while the pending request mutex is non-recursive?

Reply:
If you look at the block of code you mentioned above this, the way that block is set up, as you mentioned, the mutex is locked during `ProcessRequest`, so recursive_mutex just in case something else locks while that request is being executed.  For the normal pending requests mutex, it's just guarding the modification of that queue.  I will work on this blocking requests code to make it better.